### PR TITLE
Fix `""` escaping

### PR DESCRIPTION
**Pull request by jnewman**

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -137,7 +137,7 @@ gulp.task('test', test('coverage'));
 gulp.task('test:ci', test('ci'));
 gulp.task('test:coverage', test('coverage', false));
 gulp.task('test:cross-browser', test('sauce'));
-gulp.task('test:debug', test('coverage', true, {singleRun: false}));
+gulp.task('test:debug', test('nocoverage', true, {singleRun: false}));
```
Comment (jnewman) [inline gulpfile.babel.js]: Debugging w/ coverage is harder.

File: test/unit/utils.spec.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+describe('escapeQuotes', () => {
+  it('escapes quotes in js', () => {
+    expect(escapeQuotes('var s=""')).to.equal('var s=\\"\\"');
+    expect(escapeQuotes('"something"')).to.equal('\\"something\\"');
+    expect(escapeQuotes('"WAT')).to.equal('\\"WAT');
+  });
+
+  it("doesn't escape quotes that're already escaped", () => {
+    expect(escapeQuotes('\"WAT')).to.equal('\\"WAT');
```
Comment (jnewman) [inline test/unit/utils.spec.js]: The two backslashes seem like an implementation detail of JS strings.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "prescribe",
-  "version": "1.0.5",
+  "version": "1.0.6",
```
Comment (sethyates) [inline package.json]: Changelog update too please.

File: src/utils.js [hunk 4]
```diff
@@ -7,5 +7,9 @@
  * @returns {string}
  */
 export function escapeQuotes(value, defaultValue = '') {
-  return value ? value.replace(/(^|[^\\])"/g, '$1\\\"') : defaultValue;
+  // There's no lookback in JS, so /(^|[^\\])"/ only matches the first of two `"`s.
+  // Instead, just match anything before a double-quote and escape if it's not already escaped.
+  return !value ? defaultValue : value.replace(/(.*?)"/g, (_, prefix) => {
```
Comment (sethyates) [inline src/utils.js]: `*` and `?` are redundant, right?  `*` typically means zero or more and `?` means zero or one.  Shouldn't this just be `/(.*)"/g`?

File: test/unit/utils.spec.js (hunk 2 above)
Comment (sethyates) [inline test/unit/utils.spec.js]: Because the `\` isn't required when it's in single quotes?

Comment (jnewman) [inline test/unit/utils.spec.js]: Seems likely. I'd need to spend a good long time w/ the spec to know for sure.

File: src/utils.js (hunk 4 above)
Comment (jnewman) [inline src/utils.js]: The `*?` is less greedy. If I did `(.*)"` it'd match the second quote only.

Comment (sethyates) [inline src/utils.js]: Then is `([^"]*)"` more correct?

**Comment by sethyates** [comment]
:+1: :sparkles: :sparkles: :sparkles: :sparkles:

File: src/utils.js (hunk 4 above)
Comment (jnewman) [inline src/utils.js]: Yeah, that'll match a bit less often

**Pull request merged**